src/core/SkPicture.cpp

```diff
@@ -43,11 +43,19 @@ SkPicture::SkPicture() {
     do {
         fUniqueID = nextID.fetch_add(+1, std::memory_order_relaxed);
     } while (fUniqueID == 0);
+
+    SkRecordReplayAssert("[RUN-593-1863] SkPicture::fUniqueID %u", fUniqueID);
 }
 
 SkPicture::~SkPicture() {
     if (fAddedToCache.load()) {
-        SkResourceCache::PostPurgeSharedID(SkPicturePriv::MakeSharedID(fUniqueID));
+        if (!SkRecordReplayIsRecordingOrReplaying() || !SkRecordReplayAreEventsDisallowed()) {
+            SkResourceCache::PostPurgeSharedID(SkPicturePriv::MakeSharedID(fUniqueID));
+        }
+        else {
+            // Leak and track SkPicture id (so we get a general idea of memory impact)
+            SkRecordReplayPrint("[RUN-593-1863] SkPicture::~SkPicture - leak SkPicture %u", fUniqueID);
```

Comment:
If we see a swell in OOM or other inexplicable crashes in the near future, looking for these leak messages can be telling. This is particularly sensitive, since these images can be very big.